app/workers/generate_test_applications.rb

```diff
@@ -1,50 +1,83 @@
-    create states: %i[offer rejected]
-    create states: [:rejected] * 2
-    create states: [:offer_withdrawn]
-    create states: [:offer_deferred]
-    create states: [:offer_deferred]
-    create states: [:declined]
-    create states: [:accepted]
-    create states: [:accepted_no_conditions]
-    create states: [:recruited]
-    create states: [:conditions_not_met]
-    create states: [:withdrawn]
-    create states: [:awaiting_provider_decision], apply_again: true
+    create recruitment_cycle_year: 2020, states: %i[rejected rejected]
+    create recruitment_cycle_year: 2020, states: %i[offer_withdrawn]
+    create recruitment_cycle_year: 2020, states: %i[offer_deferred]
+    create recruitment_cycle_year: 2020, states: %i[offer_deferred]
+    create recruitment_cycle_year: 2020, states: %i[declined]
+    create recruitment_cycle_year: 2020, states: %i[accepted]
+    create recruitment_cycle_year: 2020, states: %i[recruited]
+    create recruitment_cycle_year: 2020, states: %i[conditions_not_met]
+    create recruitment_cycle_year: 2020, states: %i[withdrawn]
+    create recruitment_cycle_year: 2020, states: %i[application_not_sent]
+
+    create recruitment_cycle_year: 2021, states: %i[unsubmitted]
+    create recruitment_cycle_year: 2021, states: %i[unsubmitted], course_full: true
+    create recruitment_cycle_year: 2021, states: %i[awaiting_provider_decision awaiting_provider_decision awaiting_provider_decision]
+    create recruitment_cycle_year: 2021, states: %i[awaiting_provider_decision awaiting_provider_decision awaiting_provider_decision]
+    create recruitment_cycle_year: 2021, states: %i[awaiting_provider_decision awaiting_provider_decision awaiting_provider_decision]
+    create recruitment_cycle_year: 2021, states: %i[awaiting_provider_decision], apply_again: true
+    create recruitment_cycle_year: 2021, states: %i[offer offer]
+    create recruitment_cycle_year: 2021, states: %i[offer rejected]
+    create recruitment_cycle_year: 2021, states: %i[rejected rejected]
+    create recruitment_cycle_year: 2021, states: %i[offer_withdrawn]
+    create recruitment_cycle_year: 2021, states: %i[offer_deferred]
+    create recruitment_cycle_year: 2021, states: %i[offer_deferred]
+    create recruitment_cycle_year: 2021, states: %i[declined]
+    create recruitment_cycle_year: 2021, states: %i[accepted]
+    create recruitment_cycle_year: 2021, states: %i[accepted_no_conditions]
+    create recruitment_cycle_year: 2021, states: %i[recruited]
+    create recruitment_cycle_year: 2021, states: %i[conditions_not_met]
+    create recruitment_cycle_year: 2021, states: %i[withdrawn]
+
+    without_slack_message_sending do
+      RejectApplicationsByDefault.new.call
+    end
   end
 
-  def create(states:, apply_again: false, course_full: false)
-    @test_applications.create_application(
-      courses_to_apply_to: @courses_to_apply_to,
+private
+
+  def create(recruitment_cycle_year: nil, states:, apply_again: false, course_full: false)
```

Comment:
```suggestion
  def create(recruitment_cycle_year:, states:, apply_again: false, course_full: false)
```

Is it worth making `recruitment_cycle_year:` mandatory here, too? We've already made it mandatory for `TestApplications`.